robot/web/stream/camera/camera.go

```diff
@@ -37,9 +39,16 @@ func VideoSourceFromCamera(ctx context.Context, cam camera.Camera) gostream.Vide
 	})
 
 	img, err := camera.DecodeImageFromCamera(ctx, "", nil, cam)
-	if err == nil {
-		return gostream.NewVideoSource(reader, prop.Video{Width: img.Bounds().Dx(), Height: img.Bounds().Dy()})
+	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
+		if err := lazyImg.DecodeConfig(); err != nil {
+			return nil, fmt.Errorf("failed to decode lazy encoded image: %w", err)
+		}
 	}
```

Comment:
why are you acting on the image before you check for an error?

Shouldn't you do the error check on line 48 first & then this check?